### DISCUSSION TITLE
People:New page counts Hidden faces in the ~50 it shows

### DISCUSSION THREAD

**Discussion by rfortier** [photoprism/photoprism#3086] [Q&A]
PLEASE PROCEED ONLY IF YOU ARE ABSOLUTELY SURE THAT THIS IS NOT A TECHNICAL SUPPORT INCIDENT AND/OR POSSIBLY A PROBLEM WITH SOME OTHER SOFTWARE YOU ARE USING. VISIT <https://photoprism.app/kb/getting-support> TO LEARN MORE ABOUT OUR SUPPORT OPTIONS. THANK YOU FOR YOUR CAREFUL CONSIDERATION!

---------------------------------------------------------------------------

#### 1. What is not working as documented?

The /library/people/new page appears to be designed to show about 50 face clusters of unnamed faces. Unfortunately, it includes HIDDEN faces in that count, so once you have 50 hidden faces you can't get any new face clusters.

If it were possible to delete a face cluster entirely that would be an acceptable workaround for my particular problem. I have random background people cluttering dozens or hundreds of photos. But it appears a face can only be deleted one picture at a time, there is no way to delete the face cluster.

But the full fix should include ignoring hidden faces when displaying the New tab. With an option to include hidden faces in the list, of course. 

#### 2. How can we reproduce it?

1. Access a library with more than 50 unknown face clusters. 
2. Hide a face, several or all of them.
3. Note that no additional face clusters are displayed. 
4. This remains true when the page is refreshed, even refreshed nocache

When reporting an import, indexing, or performance issue, please include the number and type of pictures in your library, as well as any configuration options you have changed, such as for thumbnail quality.

- about 30,000 pictures.

#### 3. What behavior do you expect?

Bottom line, the display is capped at the number of faces you see when you select "show hidden." The desired behavior is to show at least that many clusters when the "exclude hidden" option is selected.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

If it becomes necessary. But I think this behavior will prove universal.

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: AMD64, ARM64, ARMv7,...

[PhotoPrism® Plus])
Build [221118-e58fee0fb]
x64 QNAP NAS, which runs a Debian variant.

(b) Database Type & Version: MariaDB, MySQL, SQLite,...
MariaDB

(c) Operating System Types & Versions: Linux, Windows, Android,...
QNAP Debian variant
uname -a
Linux NAS01 5.10.60-qnap #1 SMP Thu Dec 15 01:02:36 CST 2022 x86_64 GNU/Linux

(d) Browser Types & Versions: Firefox, Chrome, Safari on iPhone,...
Chrome 108.0.5359.125

(e) Ad Blockers, Browser Plugins, and/or Firewall Software?
Nothing relevant, but AdBlock, Windows 10 Defender firewall

#### 7. On what kind of device is PhotoPrism installed?

QNAP TS-464, 32GB. This is an x64 device.

(a) Device / Processor Type: Raspberry Pi 4, Intel Core i7-3770, AMD Ryzen 7 3800X,...
Intel Celeron N5105/N5095 quad core

(b) Physical Memory & Swap Space in GB
32GB, not swapping

(c) Storage Type: HDD, SSD, RAID, USB, Network Storage,...
RAID-5

(d) Anything else that might be helpful to know?
Installed Docker version.

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?
Yes, but same behavior when proxy bypassed.

**Comment by graciousgrey** [comment]
I cannot reproduce this.

I just tested with my own instance it has in total 62 face clusters in the new section. All except 4 of them are hidden.
As expected my instance is displaying all 62 of them when I select "​show hidden" and it is displaying 4 when I select "exclude hidden".

Is it possible that you simply have 50 hidden face clusters and all other face clusters are named already? What count do you see when you select "show hidden" and what count when you select "exclude hidden"?

**Comment by rfortier** [comment]
Maybe I wasn't clear enough in the problem statement. 

Your "As expected..." comment accurately describes the behavior, it behaves that way for me, also. I don't think that is the desired behavior, though.

Photoprism knows about many more face clusters than it shows on the library/people/new tab. The goal of that tab is to label unlabeled faces. But I can't get to the unlabeled faces to name them. 

Push the example to the limit and the problem is obvious. You currently have 62 faces, 58 are hidden. Hide the last 4 faces. Now, the New tab is completely blank and cannot be used for its intended purpose of labeling unlabeled / unknown face clusters. 

A complete fix would be a) the page should be full, regardless of whether "show hidden" is selected; b) with show hidden deselected, the page would be full of unknown faces, only those that aren't hidden; c) with show hidden selected, it would be filled with only hidden face clusters.

Then you need a way to "show more." Either a paging or endless-scroll implementation would be fine. The reasons are: a) Maybe I don't know the first 50 or so faces (because the clusters are dominated people who's names I have to get) but I know names after that; or b) I found the name of the person I hid, but they are at position 100 in the list; how do I get there?

But in the best agile tradition and for a quick fix, change the query to include filtering on "not hidden" or "hidden" selection, and always display N. Bumping N up to 100 or 200 would also be quick and an immediate improvement while a more complete fix is considered.

**Comment by graciousgrey** [comment]
> Photoprism knows about many more face clusters than it shows on the library/people/new tab. The goal of that tab is to label unlabeled faces. But I can't get to the unlabeled faces to name them.

The goal of that tab is to show you **face clusters**. Single faces that could not be matched to a **face cluster** should not show up there. 

See also: https://docs.photoprism.app/user-guide/organize/people/#why-doesnt-the-new-faces-page-show-all-faces

Improvements for the usability of the faces:new search are planned:

- #1601 
- #1679

**Comment by rfortier** [comment]
I understand the New tab shows face clusters and not individual faces; that's why I want to use it to avoid labeling individual pictures. I have 30,000 pictures, there should be many, many more face clusters. At least every other photo app has found many more. 

If it will help I'm happy to go pop into the MariaDB container and run a query to count them. You can save me time trying to figure out the query by pointing me in the right direction.

**Comment by graciousgrey** [comment]
>  At least every other photo app has found many more.

There might be plenty of reasons for this. Each app uses it's own algorithm. It might be that some apps display single faces instead of clusters or that there are more clusters because the clusters are smaller....

All face clusters found by PhotoPrism will be displayed on the Faces/New page. There is no built in limit. I just checked with another instance that displayed over 250 clusters on Faces/New.

Advanced users can play around with the facial recognition settings as explained here: https://docs.photoprism.app/getting-started/config-options/#face-recognition

**Comment by lastzero** [comment]
If your main goal is to have more clusters, you can reduce `PHOTOPRISM_FACE_CLUSTER_CORE` from 4 to 3 or 2. You can even choose 1, so that one face is one cluster. However, we do not recommend this.

**Comment by rfortier** [comment]
Thanks, those comments are really helpful. I'll give it a shot.
